be/src/formats/orc/orc_chunk_reader.cpp

```diff
@@ -1250,4 +1264,87 @@ bool OrcChunkReader::is_implicit_castable(TypeDescriptor& starrocks_type, const
     return false;
 }
 
+Status OrcChunkReader::get_schema(std::vector<SlotDescriptor>* schema) {
+    auto const& root = _reader->getType();
+
+    auto cnt = root.getSubtypeCount();
+    for (uint64_t i = 0; i < cnt; i++) {
+        TypeDescriptor tp;
+        auto name = root.getFieldName(i);
+
+        auto subtype = root.getSubtype(i);
+        switch (subtype->getKind()) {
+        case orc::TypeKind::BOOLEAN:
```

Comment:
replace with a map ?

Reply:
I don't find a good way to use a map to handle types like `orc::TypeKind::STRING`. The SR column type is created by a function call `TypeDescriptor::create_varchar_type(TypeDescriptor::MAX_VARCHAR_LENGTH)`.

Reply:
I don't find a good way to use a map to handle types like `orc::TypeKind::STRING`. The SR column type is created by a function call `TypeDescriptor::create_varchar_type(TypeDescriptor::MAX_VARCHAR_LENGTH)`.